src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -493,6 +493,22 @@ protected internal void OnOutputFilePathChanged(ProjectId projectId, string outp
             }
         }
 
+        /// <summary>
+        /// Call this method when a project's output ref file path is changed in the host environment.
+        /// </summary>
+        protected internal void OnOutputRefFilePathChanged(ProjectId projectId, string outputFilePath)
```

Comment:
Is this added speculatively, or should there be a caller already?  This change seems a bit different from the rest of the PR.

Reply:
I'm using it in my uber-commit with the big refactoring.